main.tf

```diff
@@ -80,6 +80,12 @@ locals {
         apps  = []
     }, b.restrictions)] : []
   ]
+
+  pages = {
+    branch = try(var.pages.branch, "/")
+    path   = try(var.pages.path, null)
+    cname  = try(var.pages.cname, null)
+  }
```

Comment:
please use `try()` directly in the resource no need to create local variable here and follow the `template` implementation in terms of using defaults.

Reply:
resolved in 3af6029